packages/viz/src/lib/Layer.ts

```diff
@@ -1,107 +1,107 @@
-    );
+    this._source = this._buildSource(source);
 
-    if (this._mvtLayerInstance) {
+    if (this._deckInstance) {
       await this._replaceLayer(previousSource);
     }
   }
 
+  /**
+   * Change the styles of the current layer.
+   * A new map instantion and a replace of the layer will be fired
+   * @param style style to be set
+   */
   public async setStyle(style: {}) {
-    const previousSource = this._layerSource;
+    const previousSource = this._source;
 
-    this._layerStyles = new Style(style);
+    this._styles = new Style(style);
 
-    if (this._mvtLayerInstance) {
+    if (this._deckLayer) {
       await this._replaceLayer(previousSource);
     }
   }
 
+  /**
+   * Add the current layer to a Deck instance
+   * @param deckInstance instance to add the layer to
+   */
   public async addTo(deckInstance: any) {
+    // get current layers
     const currentDeckLayers = deckInstance.props.layers;
-    const createdDeckLayer = await this.getDeckGLLayer();
+    // create the new layer
+    const layer = await this._createDeckGLLayer();
 
     deckInstance.setProps({
       layers: [
         ...currentDeckLayers,
-        createdDeckLayer
+        layer
       ]
     });
 
     this._deckInstance = deckInstance;
   }
 
-  public async getDeckGLLayer() {
-    const layerProperties = await this._getDeckGLLayerProperties();
-
-    this._mvtLayerInstance = new MVTLayer(layerProperties);
-    return this._mvtLayerInstance;
-  }
+  /**
+   * Method to create the Deck.gl layer
+   */
+  private async _createDeckGLLayer() {
 
-  private async _getDeckGLLayerProperties() {
-    const { urlTemplates: data, geometryType } = await this._layerInstantiation.then(this._parseInstantiationResult);
-    const defaultGeometryStyles = defaultStyles[geometryType];
-    const layerId = this.generateLayerId(this._layerSource.getSourceValue());
+    // Get the blueprint of the layer
+    const blueprint = await this._source.blueprint();
 
-    return Object.assign(
-      { data, id: layerId },
-      defaultGeometryStyles.getProperties(),
-      this._layerStyles.getProperties()
+    // Build the properties expected by Deck.gl
+    const layerProperties =  Object.assign(
+      { data: blueprint.tileURL, id: this._source.id },
+      defaultStyles[blueprint.geometryType].getProperties(),
+      this._styles.getProperties()
     );
+
+    // Create the Deck.gl instance
+    if (this._source instanceof CARTOSource) {
+      this._deckLayer = new MVTLayer(layerProperties);
+    } else {
+      throw Error('Unsupported source instance');
+    }
+
+    return this._deckLayer;
   }
 
-  private async _replaceLayer(previousLayerSource: Source) {
-    const newLayer = await this.getDeckGLLayer();
-    const previousLayerId = this.generateLayerId(previousLayerSource.getSourceValue());
+  /**
+   * Replace a layer source
+   * @param previousSource source of the layer to be replaced
+   */
+  private async _replaceLayer(previousSource: Source) {
```

Comment:
In this particular case, we're passing the whole `previousSource` object, but we are using it just to get its id and find the layer. I find this a bit weird, so I would propose to pass just the id. And having a look at the intention of the code, I would suggest a better naming. 

What about something like...?:

```
  private async _refreshLayer(layerId: string) {
    const newLayer = await this._createDeckGLLayer();

    const deckLayers = this._deckInstance.props.layers.filter(
      (layer: any) => layer.id !== layerId
    );

    this._deckInstance.setProps({
      layers: [
        ...deckLayers,
        newLayer
      ]
    });
  }
```

Reply:
It's just a naming discussion, but a 'replace' keeping the id is the only way to 'refresh' that deckgl exposes.